### PR TITLE
ADD: Registration controller

### PR DESCRIPTION
**Pull request by JDeepD**

### CODE REVIEW

**Review by JOSU9435 — COMMENTED** [review]
- rename the file src/controllers/registration/registration.ts to "add.ts" or "create.ts"

Since it is in the directory named registration it is understood that the controllers in this dir. are performing operations on the registration model and their file name should specify the kind of operation it's performing

- change src/middlewares/validations.ts to src/middlewares/registration/validation.ts

So that we could infer that the validtions.ts middleware is to be used for the registration routes

**Review by JOSU9435 — CHANGES_REQUESTED** [review]
- rename the file src/controllers/registration/registration.ts to "add.ts" or "create.ts"

Since it is in the directory named registration it is understood that the controllers in this dir. are performing operations on the registration model and their file name should specify the kind of operation it's performing

- change src/middlewares/validations.ts to src/middlewares/registration/validation.ts

So that we could infer that the validtions.ts middleware is to be used for the registration routes

**Comment by JDeepD** [comment]
> * rename the file src/controllers/registration/registration.ts to "add.ts" or "create.ts"
> 
> Since it is in the directory named registration it is understood that the controllers in this dir. are performing operations on the registration model and their file name should specify the kind of operation it's performing

Will do.

> * change src/middlewares/validations.ts to src/middlewares/registration/validation.ts
> 
> So that we could infer that the validtions.ts middleware is to be used for the registration routes

Can do this but `validation.ts` can actually log **any errors** (Not  just in registration). (It just logs errors that happened in previous middleware VALIDATORS). So its not  used **only** for registration validation.

**Comment by JOSU9435** [comment]
ok then no issues regarding validation.ts

File: src/middlewares/registration/checkIsTeamEvent.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    Utils.Registration.eventInfo[eventName as Utils.Registration.EventId]
+      .isTeamEvent;
+  if (!isTeamEvent && members.length > 1) {
+    return next(Utils.Response.error("Invalid Team size for this event", 400));
+  }
+
+  return next();
+};
+
+export default checkIsTeamEvent;
```
Comment (JOSU9435) [inline src/middlewares/registration/checkIsTeamEvent.ts]: if we are checking for min and max team members is there no reason to check for isTeamEvent

File: src/controllers/registration/create.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  /* If teamName is not provided(solo event), make sure member has exactly 1 entry */
+  if (!teamName && members.length > 1) {
+    return next(
+      Utils.Response.error(
+        "Cannot register more than 1 person for a solo event.",
+        400
+      )
+    );
+  }
```
Comment (JOSU9435) [inline src/controllers/registration/create.ts]: I think this check here is not required if you are checking for min and max members in middleware

**Review by JOSU9435 — CHANGES_REQUESTED** [review]

**Review by JOSU9435 — APPROVED** [review]

**Pull request merged**